Description: Some file systems, such as ext2/ext3 that were created a few years ago, do not support ACLs. As long as we do not have /media on a tmpfs, ignore failures to set the ACL on /media/<user> and fall back to chowning the directory to the target user.
Bug-Ubuntu: https://launchpad.net/bugs/1048059

Index: udisks2/src/udiskslinuxfilesystem.c
===================================================================
--- udisks2.orig/src/udiskslinuxfilesystem.c	2013-03-21 07:45:01.404209172 +0100
+++ udisks2/src/udiskslinuxfilesystem.c	2013-03-21 07:45:01.400209172 +0100
@@ -802,12 +802,10 @@
       acl_calc_mask (&acl) == -1 ||
       acl_set_file (path, ACL_TYPE_ACCESS, acl) == -1)
     {
-      g_set_error (error,
-                   G_IO_ERROR,
-                   g_io_error_from_errno (errno),
+      udisks_warning(
                    "Adding read ACL for uid %d to `%s' failed: %m",
                    (gint) uid, path);
-      goto out;
+      chown(path, uid, -1);
     }
 
   ret = TRUE;
